test/test_retriever.py

```diff
@@ -153,4 +155,4 @@ def test_format_insert_value_string_complex():
 def test_getmd5():
     """Test md5 sum calculation"""
     lines = ['a,b,c\n', '1,2,3\n', '4,5,6\n']
-    assert getmd5(lines) == '0bec5bf6f93c547bc9c6774acaf85e1a'
+    assert getmd5(lines=lines) == '0bec5bf6f93c547bc9c6774acaf85e1a'
```

Comment:
If you end up making my suggested changes to `getmd5` this call will need to be modified to:

`getmd5(data=lines, data_type='lines')